### ISSUE TITLE
API use in current version

### ISSUE THREAD

**Issue by OleVik** [COCOPi/cockpit#459]
I have searched through the issues and read the docs, but have yet to hit upon the right URL for using the Rest API. Docs example uses `/cockpit/rest/api-js?token={api_token}` with `/regions/get/{regionname}`, some issues say `cockpit/index.php/rest/api/collections/get/collectionname` or `cockpit/api/collections/get/{{collectionName}}?token={{apikey}}`.

None of these seem to work, and the example from the Docs (in JS) consequently returns a 404. As does the following:
- `http://localhost/cockpit/index.php/api/collections/get/pages?token=0432f985dbd3e29e2433042c`
- `http://localhost/cockpit/index.php/rest/api/collections/get/pages?token=0432f985dbd3e29e2433042c`

Ie, they fail with a "Uuuups, Page not found." page generated by Cockpit, whilst:
- `http://localhost/cockpit/api/collections/get/pages?token=0432f985dbd3e29e2433042c`
- `http://localhost/cockpit/rest/api/collections/get/pages?token=0432f985dbd3e29e2433042c`

Fail with a plain 404.

What is the correct URL to access the API in the current version? (v0.13.1)

**Comment by gryphonmyers** [comment]
http://{cockpit-base}/api/collections/get/{collectionName}?token={apiKey} is the correct path. If you are seeing /index.php in your paths, you don't have mod_rewrite (or its nginx equivalent) set up correctly. That could be interfering with the PHP routing. 

You should also verify that you have the correct collection name (it is case sensitive).

**Comment by OleVik** [comment]
Sorry, I forgot what environment I was in. Rather than Wamp, I was using Caddy, which does not use .htaccess-files. I switched to Wamp (Apache 2.4.17, PHP 7.0.0), verified that mod_rewrite is enabled, and am using the [current .htaccess](https://github.com/COCOPi/cockpit/blob/next/.htaccess) in `root/cockpit/cockpit/` (the address `cockpit.dev` points to `root/cockpit/`).

Both the `{collectionName}`s actual name (Pages) and its slug (pages) returns an "Uuuups, Page not found." when testing them in Postman:

- `http://cockpit.dev/cockpit/api/collections/get/Pages?token=0432f985dbd3e29e2433042c`
- `http://cockpit.dev/cockpit/api/collections/get/pages?token=0432f985dbd3e29e2433042c`

**Comment by gryphonmyers** [comment]
Two thoughts: are you sure the name of the collection is plural? Try without cockpit in the path. FYI it is looking for the collection name, not the slug

**Comment by gryphonmyers** [comment]
Another thing to check: "make also sure that $_SERVER['DOCUMENT_ROOT'] exists and is set correctly."

**Comment by OleVik** [comment]
The collection's name is pluralized as `Pages`, `$_SERVER['DOCUMENT_ROOT']` returns `C:/wamp/www/cockpit` - ie, the `root/cockpit/` ~ `cockpit.dev` folder.

I tried `http://cockpit.dev/api/collections/get/Pages?token=0432f985dbd3e29e2433042c` with Postman, and this returns `The requested URL /api/collections/get/Pages was not found on this server.`

**Comment by timaschew** [comment]
> you don't have mod_rewrite (or its nginx equivalent) set up correctly. That could be interfering with the PHP routing.

How to setup it correctly?
For me it was also not working out of the box and I'm using PHP 7.1 (CGI).
I changed the file like it was mentioned here in this comment: https://github.com/COCOPi/cockpit/issues/350#issuecomment-130819975

After that the routes didn't contain `index.php` anymore.

---

I've realized that the slug of the collection is not working for the rest API, I was trying to use it all the time. In my case the difference was just one capital letter.

The API URL for me is this:

`http://cms.mydomain.tld/rest/api/collections/get/COLLECTION_NAME?token=MY_TOKEN`

And the webinterface is under `http://cms.mydomain.tld`
